### PR TITLE
[LOGS] Improve zip error logs

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Improve zip error logs
| Type?             | improvement
| BC breaks?        |no
| Deprecations?     |no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | -

### CODE REVIEW

**Comment by M0rgan01** [comment]
Proposal for improving error logs during zip actions

File: classes/ZipAction.php [hunk 1]
```diff
@@ -159,10 +166,15 @@ public function extract(string $from_file, string $to_dir): bool
 
         for ($i = 0; $i < $zip->numFiles; ++$i) {
             if (!$zip->extractTo($to_dir, [$zip->getNameIndex($i)])) {
+                $error = $zip->getStatusString();
+
                 $this->logger->error(
                     $this->translator->trans(
-                        'Could not extract %file% from backup, the destination might not be writable.',
-                        ['%file%' => $zip->statIndex($i)['name']]
+                        'Could not extract %file%: %error%',
```
Comment (Quetzacoalt91) [inline classes/ZipAction.php]:
```suggestion
                        'Could not extract %file% from archive: %error%',
```

File: classes/ZipAction.php [hunk 2]
```diff
@@ -257,8 +269,10 @@ public function open(string $zipFile, ?int $flags = null): ZipArchive
             $flags = 0;
         }
         if ($zip->open($zipFile, $flags) !== true || empty($zip->filename)) {
-            $this->logger->error($this->translator->trans('Unable to open zipFile %s', [$zipFile]));
-            throw new ZipActionException('Unable to open zipFile ' . $zipFile);
+            $error = $zip->getStatusString();
+
+            $this->logger->error($this->translator->trans('Unable to open zipFile %s: %s', [$zipFile, $error]));
```
Comment (Quetzacoalt91) [inline classes/ZipAction.php]:
```suggestion
            $this->logger->error($this->translator->trans('Unable to open archive %s: %s', [$zipFile, $error]));
```

File: classes/ZipAction.php [hunk 3]
```diff
@@ -257,8 +269,10 @@ public function open(string $zipFile, ?int $flags = null): ZipArchive
...
+            throw new ZipActionException('Unable to open zipFile ' . $zipFile . ': ' . $error);
```
Comment (Quetzacoalt91) [inline classes/ZipAction.php]:
```suggestion
            throw new ZipActionException('Unable to open archive ' . $zipFile . ': ' . $error);
```

**Review by ga-devfront — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as you can see : 

image

image

------------

image

image

----------------------

image

image

--------------------

image


image



Tested from : 
1.7.8.8 to 8.2 
8.0.4 to 8.2 
8.2 to 9.0.0
8.0.4 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**